### PR TITLE
cherrypick SN service annotation PRs to master

### PR DESCRIPTION
**Pull request by dunjut**
#8642

### CODE REVIEW

**Comment by dunjut** [comment]
cc @quanjielin @lei-tang

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -412,6 +412,7 @@ pilot:
       memory: 2048Mi
   env:
     PILOT_PUSH_THROTTLE_COUNT: 100
+    ISTIO_SA_DOMAIN_CANONICAL: accounts.google.com
```
[the remark this answers is not in the document]
Comment (quanjielin) [reply install/kubernetes/helm/istio/values.yaml]: @dunjut thanks for the change;  value is defined and set to be "" in this file, we could have a separated -values.yaml file overrides this to be ```accounts.google.com``` later;

File: pilot/pkg/serviceregistry/kube/conversion.go [hunk 2]
```diff
@@ -47,6 +48,20 @@ const (
 	managementPortPrefix = "mgmt-"
 )
 
+var (
+	// istioServiceAccountDomainForCanonical is the domain in the Istio service account scheme
+	// when converting a Canonical service account
+	// Default value is "accounts.google.com", or the value of ISTIO_SA_DOMAIN_CANONICAL env
+	istioServiceAccountDomainForCanonical = "accounts.google.com"
```
Comment (quanjielin) [inline pilot/pkg/serviceregistry/kube/conversion.go]: set default value to be ""

**Comment by dunjut** [comment]
@linsun @quanjielin Sorry for setting google value as default. Just pushed a new commit, please point out if it still need any further fix.

**Comment by quanjielin** [comment]
/lgtm 
@lei-tang could you pls take a look at this to make sure nothing is missed from your previous PRs ?

**Comment by lei-tang** [comment]
The git command to cherry-pick a commit is: 
git cherry-pick HASH-OF-THE-COMMIT-TO-CHERRY-PICK

**Comment by lei-tang** [comment]
It should be ok to make force pushes. As the subject of this PR is cherrypicking two existing commits, you may make a new PR to remove hardcoded google value and commit the new PR after this cherrypicking PR is merged.

**Comment by dunjut** [comment]
@lei-tang Please review again ;-)

**Comment by lei-tang** [comment]
There are circleci build and lint errors. Please add a new commit to fix them.

**Comment by dunjut** [comment]
@lei-tang Fixed

**Comment by lei-tang** [comment]
Thanks! /lgtm

File: pilot/pkg/serviceregistry/kube/conversion.go [hunk 3]
```diff
@@ -125,7 +125,7 @@ func serviceHostname(name, namespace, domainSuffix string) model.Hostname {
 
 // canonicalToIstioServiceAccount converts a Canonical service account to an Istio service account
 func canonicalToIstioServiceAccount(saname string) string {
-	return fmt.Sprintf("%v://%v", istioURIPrefix, saname)
+	return fmt.Sprintf("%v://accounts.google.com/%v", istioURIPrefix, saname)
```
Comment (rshriram) [inline pilot/pkg/serviceregistry/kube/conversion.go]: why the accounts.google.com ? whats the relevance?

Comment (dunjut) [reply pilot/pkg/serviceregistry/kube/conversion.go]: The original author @lei-tang [suggests](https://github.com/istio/istio/pull/8657#issuecomment-420864129): `As the subject of this PR is cherrypicking two existing commits, you may make a new PR to remove hardcoded google value and commit the new PR after this cherrypicking PR is merged.`

Now I'm not sure whether I should remove the hardcoded google value in this PR, or in another new PR?

Comment (lei-tang) [reply pilot/pkg/serviceregistry/kube/conversion.go]: I am ok with removing the hardcoded google value in this PR, e.g., you can revise your commit 07dd182 "fix wrong variable name" to remove the hardcoded google value.

**Comment by myidpt** [comment]
Thanks for the work!

**Pull request merged**